src/main/java/pl/jablonskanycz/bakery/clients/FileBasedClientRepository.java

```diff
@@ -1,30 +1,127 @@
 package pl.jablonskanycz.bakery.clients;
 
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
 import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static java.nio.file.StandardOpenOption.APPEND;
 
 public class FileBasedClientRepository implements ClientRepository {
+    private static final Path clientPath = Path.of("src", "main", "resources", "CLIENT.csv");
+    private AddressRepository addressRepository = new FileBasedAddressRepository();
     @Override
     public List<Client> getAll() {
-        return null;
+            List<Client> clients = null;
+            try {
+                clients = Files.lines(clientPath)
+                .skip(1)
+                .map(line -> {
+                    String[] client = line.split(",");
+                    return new Client(
+                        Integer.parseInt(client[0]),
+                        client[1],
+                        client[2],
+                        addressRepository.findById(Integer.parseInt(client[3])));
+                })
+                .collect(Collectors.toList());
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+            return clients;
     }
 
     @Override
     public Client findBySurname(String surname) {
-        return null;
+        Client client = null;
+        try {
+            return Files.lines(clientPath)
+                    .skip(1)
+                    .map(line -> {
+                        String[] strings = line.split(",");
+                        return new Client(
+                                Integer.parseInt(strings[0]),
+                                strings[1],
+                                strings[2],
+                                addressRepository.findById(Integer.parseInt(strings[3])));
+                    })
+                    .filter(a -> surname.equals(a.getSurname()))
+                    .findFirst()
+                    .orElseThrow(() -> new NoSuchElementException("Client list is empty"));
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return client;
     }
 
     @Override
     public void addClient(Client clientToAdd) {
-
+        List<String> clientsList = null;
```

Comment:
we need this list to be initialized here - that's correct! 
It will be later pointed to the result of reading the file. 
In happy case, all will be fine and the list will indeed consist of `Strings`. 
However, when for some reason the reading from the file fails - the code will move on to lie 67th and then... to line 69. With the `clientsList` still being `null`, it will then call `clientsList.get(..)` which would then throw `NullPointerException`. 

How could we prevent this case:)? What would you suggest? :)